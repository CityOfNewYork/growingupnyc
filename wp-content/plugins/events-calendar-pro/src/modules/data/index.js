/**
 * External dependencies
 */
import reducer from './reducers';

import { actions, constants } from '@moderntribe/common/data/plugins';
import { store } from '@moderntribe/common/store';
import initSagas from './sagas';
import * as blocks from './blocks';
import '@moderntribe/events-pro/data/sagas';

const { EVENTS_PRO_PLUGIN } = constants;

export const initStore = () => {
	const { dispatch, injectReducers } = store;
	initSagas();

	dispatch( actions.addPlugin( EVENTS_PRO_PLUGIN ) );
	injectReducers( { [ EVENTS_PRO_PLUGIN ]: reducer } );
};

export const getStore = () => store;

export { blocks };
