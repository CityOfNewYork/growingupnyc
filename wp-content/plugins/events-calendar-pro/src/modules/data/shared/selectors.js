/* eslint-disable max-len */

/**
 * External dependencies
 */
import { createSelector } from 'reselect';
import * as constants from '@moderntribe/events-pro/data/blocks/constants';

export const getRule = rule => rule;

export const getType = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_TYPE ],
);

export const getAllDay = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_ALL_DAY ],
);

export const getMultiDay = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_MULTI_DAY ],
);

export const getMultiDaySpan = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_MULTI_DAY_SPAN ],
);

export const getStartDate = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_START_DATE ],
);

export const getStartDateInput = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_START_DATE_INPUT ],
);

export const getStartDateObj = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_START_DATE_OBJ ],
);

export const getStartTime = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_START_TIME ],
);

export const getStartTimeNoSeconds = createSelector(
	[ getStartTime ],
	( startTime ) => startTime.slice( 0, -3 ),
);

export const getStartTimeInput = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_START_TIME_INPUT ],
);

export const getEndDate = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_END_DATE ],
);

export const getEndDateInput = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_END_DATE_INPUT ],
);

export const getEndDateObj = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_END_DATE_OBJ ],
);

export const getEndTime = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_END_TIME ],
);

export const getEndTimeNoSeconds = createSelector(
	[ getEndTime ],
	( endTime ) => endTime.slice( 0, -3 ),
);

export const getEndTimeInput = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_END_TIME_INPUT ],
);

export const getBetween = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_BETWEEN ],
);

export const getLimitType = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_LIMIT_TYPE ],
);

export const getLimit = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_LIMIT ],
);

export const getLimitDateInput = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_LIMIT_DATE_INPUT ],
);

export const getLimitDateObj = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_LIMIT_DATE_OBJ ],
);

export const getDays = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_DAYS ],
);

export const getWeek = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_WEEK ],
);

export const getDay = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_DAY ],
);

export const getMonth = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_MONTH ],
);

export const getTimezone = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_TIMEZONE ],
);
