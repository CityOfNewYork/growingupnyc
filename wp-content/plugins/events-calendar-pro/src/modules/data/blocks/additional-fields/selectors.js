/**
 * External dependencies
 */
import { createSelector } from 'reselect';
import identity from 'lodash/identity';
import includes from 'lodash/includes';
import uniq from 'lodash/uniq';

/**
 * Internal dependencies
 */
import { constants } from '@moderntribe/common/data/plugins';

export const getPlugin = ( state ) => state[ constants.EVENTS_PRO_PLUGIN ];
export const getBlocks = createSelector(
	[ getPlugin ],
	( plugin ) => plugin.blocks,
);

export const getAdditionalFields = createSelector(
	[ getBlocks ],
	( blocks ) => blocks.additionalFields,
);

export const getFieldName = ( state, props ) => props.name;

export const getIds = createSelector(
	[ getAdditionalFields ],
	( fields ) => fields.allIds,
);

export const getFieldAsObjects = createSelector(
	[ getAdditionalFields ],
	( fields ) => fields.byId,
);

export const getFieldBlock = createSelector(
	[ getFieldAsObjects, getFieldName ],
	( fields, name ) => fields[ name ] || {},
);

export const getFieldDividerList = createSelector(
	[ getFieldBlock ],
	( field ) => field.dividerList,
);

export const getFieldDividerEnd = createSelector(
	[ getFieldBlock ],
	( field ) => field.dividerEnd,
);

export const getFieldOutput = createSelector(
	[ getFieldBlock ],
	( field ) => field.output,
);

export const getFieldMetaKey = createSelector(
	[ getFieldBlock ],
	( field ) => field.metaKey,
);

export const getFieldType = createSelector(
	[ getFieldBlock ],
	( field ) => field.type,
);

export const getFieldLabel = createSelector(
	[ getFieldBlock ],
	( field ) => field.label || '',
);

export const getFieldValue = createSelector(
	[ getFieldBlock ],
	( field ) => field.value,
);

export const getTextFieldValue = createSelector(
	[ getFieldValue ],
	( value ) => value || '',
);

export const getTextAreaOutput = createSelector(
	[ getFieldOutput ],
	( output = '' ) => {
		return output.split( '\n' ).filter( identity );
	},
);

export const getFieldIsPristine = createSelector(
	[ getFieldBlock ],
	( field ) => field.isPristine,
);

export const getFieldOptions = createSelector(
	[ getFieldBlock ],
	( field ) => field.options || [],
);

export const getFieldOptionsWithLabels = createSelector(
	[ getFieldOptions ],
	( options ) => {
		return options.map( ( option ) => ( { value: option, label: option } ) );
	},
);

export const getFieldDropdownValue = createSelector(
	[ getFieldBlock ],
	( field ) => ( { value: field.value, label: field.value } ),
);

export const getFieldDropdownOutput = createSelector(
	[ getFieldBlock ],
	( field ) => field.output,
);

export const getFieldCheckboxValue = createSelector(
	[ getTextFieldValue ],
	( value ) => uniq( value.split( '|' ) ),
);

export const getFieldCheckboxOptions = createSelector(
	[ getFieldCheckboxValue, getFieldOptionsWithLabels ],
	( values, optionsWithLabels ) => {
		return optionsWithLabels.map( ( option ) => ( {
			...option,
			isChecked: includes( values, option.value ),
		} ) );
	},
);
