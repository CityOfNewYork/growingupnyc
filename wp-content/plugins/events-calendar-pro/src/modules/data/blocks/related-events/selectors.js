/**
 * External dependencies
 */
import { createSelector } from 'reselect';
import { constants } from '@moderntribe/common/data/plugins';

export const getRelatedEventsBlock = ( state ) => state[ constants.EVENTS_PRO_PLUGIN ].blocks.relatedEvents;

export const getTitle = createSelector(
	[ getRelatedEventsBlock ],
	( relatedEvents ) => relatedEvents.title,
);

export const getDisplayImages = createSelector(
	[ getRelatedEventsBlock ],
	( relatedEvents ) => relatedEvents.displayImages,
);
