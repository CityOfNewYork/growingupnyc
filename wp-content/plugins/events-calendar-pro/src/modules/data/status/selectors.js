/**
 * External dependencies
 */
import { createSelector } from 'reselect';
import { constants } from '@moderntribe/common/data/plugins';

export const getStatus = ( state ) => state[ constants.EVENTS_PRO_PLUGIN ].status;

export const isCompleted = createSelector( getStatus, status => !! status.done );
export const getProgress = createSelector( getStatus, status => status.progress );
