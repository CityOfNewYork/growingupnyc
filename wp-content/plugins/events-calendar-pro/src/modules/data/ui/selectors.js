/**
 * External dependencies
 */
import { createSelector } from 'reselect';
import { constants } from '@moderntribe/common/data/plugins';

export const getUI = ( state ) => state[ constants.EVENTS_PRO_PLUGIN ].ui;

export const isRepeatBlockVisible = createSelector(
	[ getUI ],
	( ui ) => ui.isRepeatBlockVisible
);
export const isRulePanelVisible = createSelector(
	[ getUI ],
	( ui ) => ui.isRulePanelVisible
);
export const isExceptionPanelVisible = createSelector(
	[ getUI ],
	( ui ) => ui.isExceptionPanelVisible
);
export const isRulePanelExpanded = createSelector(
	[ getUI ],
	( ui ) => ui.isRulePanelExpanded
);
export const isExceptionPanelExpanded = createSelector(
	[ getUI ],
	( ui ) => ui.isExceptionPanelExpanded
);
