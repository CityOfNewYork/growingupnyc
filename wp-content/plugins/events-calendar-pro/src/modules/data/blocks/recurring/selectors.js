/* eslint-disable max-len */

/**
 * External dependencies
 */
import { createSelector } from 'reselect';
import { find } from 'lodash';

/**
 * Internal dependencies
 */
import {
	RECURRENCE_TYPE_RULES_OPTIONS,
	SERIES_ENDS_OPTIONS,
} from './options';
import { constants } from '@moderntribe/common/data/plugins';
import * as selectors from '@moderntribe/events-pro/data/shared/selectors';

export const getRules = ( state ) => state[ constants.EVENTS_PRO_PLUGIN ].blocks.recurring;
export const getRulesCount = createSelector( getRules, ( rules ) => rules.length );
export const hasRules = createSelector( getRulesCount, ( count ) => !! count );
export const getIndex = ( _, props ) => props.index;

export const getRule = createSelector(
	[ getRules, getIndex ],
	( rules, index ) => rules[ index ],
);

export const getType = createSelector( getRule, selectors.getType );
export const getAllDay = createSelector( getRule, selectors.getAllDay );
export const getMultiDay = createSelector( getRule, selectors.getMultiDay );
export const getMultiDaySpan = createSelector( getRule, selectors.getMultiDaySpan );
export const getStartDate = createSelector( getRule, selectors.getStartDate );
export const getStartDateObj = createSelector( getRule, selectors.getStartDateObj );
export const getStartDateInput = createSelector( getRule, selectors.getStartDateInput );
export const getStartTime = createSelector( getRule, selectors.getStartTime );
export const getStartTimeNoSeconds = createSelector( getRule, selectors.getStartTimeNoSeconds );
export const getStartTimeInput = createSelector( getRule, selectors.getStartTimeInput );
export const getEndDate = createSelector( getRule, selectors.getEndDate );
export const getEndDateObj = createSelector( getRule, selectors.getEndDateObj );
export const getEndDateInput = createSelector( getRule, selectors.getEndDateInput );
export const getEndTime = createSelector( getRule, selectors.getEndTime );
export const getEndTimeNoSeconds = createSelector( getRule, selectors.getEndTimeNoSeconds );
export const getEndTimeInput = createSelector( getRule, selectors.getEndTimeInput );
export const getBetween = createSelector( getRule, selectors.getBetween );
export const getLimitType = createSelector( getRule, selectors.getLimitType );
export const getLimit = createSelector( getRule, selectors.getLimit );
export const getLimitDateObj = createSelector( getRule, selectors.getLimitDateObj );
export const getLimitDateInput = createSelector( getRule, selectors.getLimitDateInput );
export const getDays = createSelector( getRule, selectors.getDays );
export const getDay = createSelector( getRule, selectors.getDay );
export const getMonth = createSelector( getRule, selectors.getMonth );
export const getWeek = createSelector( getRule, selectors.getWeek );
export const getTimezone = createSelector( getRule, selectors.getTimezone );

export const getTypeOption = createSelector(
	[ getType ],
	( type ) => find( RECURRENCE_TYPE_RULES_OPTIONS, ( option ) => option.value === type ),
);

export const getLimitTypeOption = createSelector(
	[ getLimitType ],
	( limitType ) => find( SERIES_ENDS_OPTIONS, ( option ) => option.value === limitType ),
);
