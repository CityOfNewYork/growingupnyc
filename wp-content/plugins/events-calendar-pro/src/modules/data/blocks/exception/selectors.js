/**
 * External dependencies
 */
import { createSelector } from 'reselect';
import { find } from 'lodash';

/**
 * Internal dependencies
 */
import {
	EXCEPTION_OCCURRENCE_OPTIONS,
} from './options';
import { SERIES_ENDS_OPTIONS } from '@moderntribe/events-pro/data/blocks/recurring/options';
import { constants } from '@moderntribe/common/data/plugins';
import * as selectors from '@moderntribe/events-pro/data/shared/selectors';

export const getExceptions = ( state ) => state[ constants.EVENTS_PRO_PLUGIN ].blocks.exception;
export const getRules = getExceptions;
export const getExceptionsCount = createSelector( getExceptions, ( exceptions ) => exceptions.length );
export const hasExceptions = createSelector( getExceptionsCount, ( count ) => !! count );
export const getIndex = ( _, props ) => props.index;

export const getRule = createSelector(
	[ getExceptions, getIndex ],
	( exceptions, index ) => exceptions[ index ],
);

export const getType = createSelector( getRule, selectors.getType );
export const getAllDay = createSelector( getRule, selectors.getAllDay );
export const getMultiDay = createSelector( getRule, selectors.getMultiDay );
export const getMultiDaySpan = createSelector( getRule, selectors.getMultiDaySpan );
export const getStartDate = createSelector( getRule, selectors.getStartDate );
export const getStartDateObj = createSelector( getRule, selectors.getStartDateObj );
export const getStartDateInput = createSelector( getRule, selectors.getStartDateInput );
export const getStartTime = createSelector( getRule, selectors.getStartTime );
export const getStartTimeNoSeconds = createSelector( getRule, selectors.getStartTimeNoSeconds );
export const getStartTimeInput = createSelector( getRule, selectors.getStartTimeInput );
export const getEndDate = createSelector( getRule, selectors.getEndDate );
export const getEndDateObj = createSelector( getRule, selectors.getEndDateObj );
export const getEndDateInput = createSelector( getRule, selectors.getEndDateInput );
export const getEndTime = createSelector( getRule, selectors.getEndTime );
export const getEndTimeNoSeconds = createSelector( getRule, selectors.getEndTimeNoSeconds );
export const getEndTimeInput = createSelector( getRule, selectors.getEndTimeInput );
export const getBetween = createSelector( getRule, selectors.getBetween );
export const getLimitType = createSelector( getRule, selectors.getLimitType );
export const getLimit = createSelector( getRule, selectors.getLimit );
export const getLimitDateObj = createSelector( getRule, selectors.getLimitDateObj );
export const getLimitDateInput = createSelector( getRule, selectors.getLimitDateInput );
export const getDays = createSelector( getRule, selectors.getDays );
export const getDay = createSelector( getRule, selectors.getDay );
export const getMonth = createSelector( getRule, selectors.getMonth );
export const getWeek = createSelector( getRule, selectors.getWeek );
export const getTimezone = createSelector( getRule, selectors.getTimezone );

export const getTypeOption = createSelector(
	[ getType ],
	( type ) => find( EXCEPTION_OCCURRENCE_OPTIONS, ( option ) => option.value === type ),
);

export const getLimitTypeOption = createSelector(
	[ getLimitType ],
	( limitType ) => find( SERIES_ENDS_OPTIONS, ( option ) => option.value === limitType ),
);
