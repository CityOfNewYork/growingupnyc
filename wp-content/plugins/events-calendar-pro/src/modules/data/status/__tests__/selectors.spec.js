/**
 * External Dependencies
 */
import * as selectors from '../selectors';
import { constants } from '@moderntribe/common/data/plugins';

describe( 'Status selectors', () => {
	let state;
	beforeEach( () => {
		state = {
			[ constants.EVENTS_PRO_PLUGIN ]: {
				status: {
					done: true,
					progress: 60,
				},
			},
		};
	} );

	const keys = Object.keys( selectors );

	keys.forEach( ( key ) => {
		test( key, () => {
			expect( selectors[ key ]( state ) ).toMatchSnapshot();
		} );
	} );
} );
