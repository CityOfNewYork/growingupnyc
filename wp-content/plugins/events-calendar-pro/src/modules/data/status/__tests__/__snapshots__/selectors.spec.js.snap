// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status selectors getProgress 1`] = `60`;

exports[`Status selectors getStatus 1`] = `
Object {
  "done": true,
  "progress": 60,
}
`;

exports[`Status selectors isCompleted 1`] = `true`;
